modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -155,7 +155,7 @@
 		return serviceRegistrations;
 	}
 
-	public ObjectDefinition getObjectDefinition(
+	public synchronized ObjectDefinition getObjectDefinition(
```

Comment:
I don't like this one... because it's getting called in createContext of `ObjectDefinitionContextProvider` that is getting called in **every** API call. So basically we are queuing every API call that will have to pass through that method in order.

I prefer to fail than adding so much contingency.

There are 2 things to consider here, and I understand why brian wants to add synchronization...
1) Is the API going to fail if the API is undeployed and the method is not synchronized? yes. Is the API going to fail if a new API is being deployed? if it doesn't share the same restContextName... NO. If it shares it? I don't think so either.
2) Can we expose a different ObjectDefinition that I don't own when an API is deployed/undeployed? I don't think so but we have to think about it. The internal map is scoped by company so I don't see how an evil deployer could do it but maybe I'm missing something.

So if the only issue is failing when the API is being undeployed, I think it's tolerable. I would remove this `synchronized`.

Reply:
I agree. The critical issue would be to expose an ObjectDefinition from another company and I think that it is not possible as we control the logic to get the companyId and select the right Object Definition.

I'll remove the synchronized keyword in the get method and push the changes

Reply:
👍